deepchecks/utils/ipython.py

```diff
@@ -116,39 +129,189 @@ def is_widgets_use_possible() -> bool:
     )
 
 
-class ProgressBar:
-    """Progress bar for display while running suite.
+def create_progress_bar(
+    iterable: t.Sequence[t.Any],
+    name: str,
+    unit: str,
+):
+    """Create a tqdm progress bar instance."""
+    kwargs = {
+        'iterable': iterable,
+        'desc': name,
+        'unit': f' {unit}',
+        'leave': True,
+    }
 
-    Parameters
-    ----------
-    name : str
-    length : int
-    """
+    iterlen = len(iterable)
+    barlen = iterlen if iterlen > 5 else 5
+
+    if is_widgets_enabled():
+        return tqdm_notebook(
+            **kwargs,
+            colour='#9d60fb',
+            file=sys.stdout
+        )
+
+    elif is_notebook():
+
+        class PB(tqdm.tqdm):
+            """Custom progress bar."""
+
+            def __init__(self, **kwargs):
+                self.display_handler = display({'text/plain': ''}, raw=True, display_id=True)
+                kwargs['file'] = io.StringIO()
+                super().__init__(**kwargs)
+
+            def refresh(self, nolock=False, lock_args=None):
+                value = super().refresh(nolock, lock_args)
+                self.display_handler.update({'text/plain': self.fp.getvalue()}, raw=True)
+                self.fp.seek(0)
+                return value
+
+            def close(self, *args, **kwargs):
+                value = super().close(*args, **kwargs)
+                self.display_handler.update({'text/plain': self.fp.getvalue()}, raw=True)
+                self.fp.seek(0)
+                return value
+
+        return PB(
+            **kwargs,
+            bar_format='{{desc}}:\n|{{bar:{0}}}{{r_bar}}'.format(barlen),  # pylint: disable=consider-using-f-string
+        )
+
+    else:
+        return tqdm.tqdm(
+            **kwargs,
+            bar_format='{{desc}}:\n|{{bar:{0}}}{{r_bar}}'.format(barlen),  # pylint: disable=consider-using-f-string
+        )
+
+
+@contextlib.contextmanager
+def create_dummy_progress_bar(name: str, unit: str):
+    """Create a tqdm progress bar instance that has only one step."""
+    for _ in create_progress_bar(
+        list(range(1)),
+        name=name,
+        unit=unit
+    ):
+        yield
+
+
+# TODO:
+# NOTE:
```

Comment:
why is it commented?

Reply:
I will create another PR to finish that

in general, I need to find a way to determine which jupyter-server extensions are enabled in order to make it work correctly, currently, I cannot find how to do that